### DISCUSSION TITLE
How to specify HTTP response status for custom endpoints?

### DISCUSSION THREAD

**Discussion by shahin-y** [atoti/atoti#568] [Q&A] [answered]
Hi there,

When adding custom endpoints, as described in the [documentation](https://docs.atoti.io/latest/lib/atoti/atoti.session.html#atoti.Session.endpoint), is it possible to specify the HTTP response code?
From the documentation ("The decorated function must [...] return a response body as a Python data structure that can be converted to JSON.") and the way the `_call_proxy` method processes the `EndpointHandler`'s return value, it seems that only the message body can be set. However, in some cases it would be quite helpful to be able to set the response code.

Is this correct or did I miss something in the documentation?

I was hoping to find something similar to the [Flask Response object](https://flask.palletsprojects.com/en/2.0.x/quickstart/#about-responses), for instance like this:

```python
@session.endpoint("services/brew-coffee", method="POST")
def brew_coffee(request, user, session):
    if request.body["tool"] == "teapot":
        return "Attempt to brew coffee with a teapot failed.", 418
    return "Coffee is ready."
```

**Comment by tibdex** [comment, marked as answer]
I agree that this would be a good improvement. It's not supported at the moment.

**Reply by shahin-y** [reply]
Alright, thanks for the quick reply!
